servicetalk-traffic-resilience-http/src/main/java/io/servicetalk/traffic/resilience/http/AbstractTrafficResilienceHttpFilter.java

```diff
@@ -332,38 +332,37 @@ public CapacityLimiter.LimiterState state() {
 
         @Override
         public int completed() {
-            signal(SIGNAL_COMPLETED);
-            return delegate.completed();
+            return signal(SIGNAL_COMPLETED) ? delegate.completed() : -1;
         }
 
         @Override
         public int dropped() {
-            signal(SIGNAL_DROPPED);
-            return delegate.dropped();
+            return signal(SIGNAL_DROPPED) ? delegate.dropped() : -1;
         }
 
         @Override
         public int failed(final Throwable error) {
-            signal(SIGNAL_FAILED);
-            return delegate.failed(error);
+            return signal(SIGNAL_FAILED) ? delegate.failed(error) : -1;
         }
 
         @Override
         public int ignored() {
-            signal(SIGNAL_IGNORED);
-            return delegate.ignored();
+            return signal(SIGNAL_IGNORED) ? delegate.ignored() : -1;
```

Comment:
IIUC -1 in this context means no more capacity, does that hold true "just because" there was a double signal?

Reply:
The return value is intended to be a hint and only if it's positive so, at least according to the API docs, negative numbers should mean 'no hint' and be ignored. As far as I can tell, the API docs are actually wrong and should say 'non-negative', but the one place in main that is concerned with the result (CompsoiteCapacityLimiter) ignores negative results.